### PR TITLE
Middle click on groupbar closes tab

### PR DESCRIPTION
**Pull request by dranull**
#### Describe your PR, what does it fix/add?
Plus some other fixes that I've noticed.

#### Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)
No.

#### Is it ready for merging, or does it need work?
Yes.

### CODE REVIEW

**Review by vaxerski — APPROVED** [review]
this looks like something I'd merge in about 12s

**Comment by vaxerski** [comment]
oops you still committing, is this ready?

**Comment by dranull** [comment]
I had a typo in the commit message...

**Comment by MightyPlaza** [comment]
it's probably a good idea to make this configurable

**Comment by dranull** [comment]
@MightyPlaza I though about that, but middle click is common for closing tabs. Also who uses middle click to select tabs (or anything)?

**Comment by MightyPlaza** [comment]
> @MightyPlaza I though about that, but middle click is common for closing tabs.

never noticed this

it's not that hard either to add it later on if someone wants it

File: src/render/decorations/CHyprGroupBarDecoration.cpp [hunk 1]
```diff
@@ -386,12 +386,28 @@ bool CHyprGroupBarDecoration::onEndWindowDragOnDeco(const Vector2D& pos, CWindow
...
+        if (e->state == WLR_BUTTON_PRESSED)
+            pressedCursorPos = pos;
+        else if (e->state == WLR_BUTTON_RELEASED && pressedCursorPos == pos)
+            g_pXWaylandManager->sendCloseWindow(m_pWindow->getGroupWindowByIndex(WINDOWINDEX));
+
+        return true;
+    }
+
+    if (e->state != WLR_BUTTON_PRESSED)
+        return true;
```
Comment (MightyPlaza) [inline src/render/decorations/CHyprGroupBarDecoration.cpp:408]: should this not be `false`?

File: src/render/decorations/CHyprGroupBarDecoration.cpp [hunk 2]
```diff
@@ -403,8 +419,8 @@ bool CHyprGroupBarDecoration::onMouseButtonOnDeco(const Vector2D& pos, wlr_point
     if (pWindow != m_pWindow)
         pWindow->setGroupCurrent(pWindow);
 
-    if (!g_pCompositor->isWindowActive(pWindow))
-        g_pCompositor->focusWindow(pWindow);
```
Comment (MightyPlaza) [inline src/render/decorations/CHyprGroupBarDecoration.cpp:407]: why was this removed?

Comment (dranull) [reply src/render/decorations/CHyprGroupBarDecoration.cpp:407]: Seemed unnecessary, since `setGroupCurrent` already called `focusWindow`. I probably removed it because it worked the same way without it while I was testing. But I don't really remember.

File: src/render/decorations/CHyprGroupBarDecoration.cpp (hunk 1 above)
Comment (dranull) [reply src/render/decorations/CHyprGroupBarDecoration.cpp:408]: This way the event won't be propagated downwards. I expect only the deco I see to handle my click. At least that was my thinking.
